File: C++/src/Calcular.cpp
#include "Calcular.h"

Calcular::Calcular()
{
    //a la variable num se le asigna 0
    //en el constructor de la clase.
    num = 0;
}

Calcular::~Calcular()
{
    //dtor
}

//implementacion del metodo cuadrado()
int Calcular::cuadrado()
{
    //se retorna la variable num multiplicada
    //dos veces
    return num * num;
}

//implementacion del metodo cubo()
int Calcular::cubo()
{
    //se retorna la variable num multiplicada
    //tres veces
    return num * num * num;
}

//implementacion del metodo modificador del
//atributo "num"
void Calcular::setNum(int val)
{
    //a "num" se le asigna "val"
    num = val;
}

//implementacion del metodo selector del
//atributo "num"
int Calcular::getNum()
{
    //se retorna "num"
    return num;
}

File: Java/src/Calcular.java
public class Calcular
{
    //Se define el atributo privado num de tipo entero
    private int num;
    
    //Se definen los metodos publicos para calcular:
    //el cuadrado y el cubo.
    public int cuadrado()
    {
        return num * num;
    }
    public int cubo()
    {
        return num * num * num;
    }

    //Se definen los metodos Setters y Getters
    //(Encapsulamiento)del atributo "num"
    public void setNum(int val)
    {
        num = val;
    }
    public int getNum()
    {
        return num;
    }
}